gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/util.py

```diff
@@ -96,13 +96,69 @@ def get_ovr_count(filename_or_ds: PathOrDS):
         return bnd.GetOverviewCount()
 
 
-def get_ovr_idx(filename_or_ds: PathOrDS, ovr_idx: Optional[int]):
+def get_sizes_factors_resolutions(filename_or_ds: PathOrDS, dim: Optional[int]=0):
+    ds = open_ds(filename_or_ds)
+    bnd = ds.GetRasterBand(1)
+    ovr_count = bnd.GetOverviewCount()
+    geo_transform = ds.GetGeoTransform()
```

Comment:
Use ds.GetGeoTransform(can_return_null=True) and deal with the case when geo_transform is None

Reply:
done